### PR TITLE
Theme fixes for proposition pages (and elsewhere

### PR DESCRIPTION
**Pull request by helenb**
Theme fixes for proposition pages (and elsewhere), and remove the option to add the dark transparent theme for proposition pages

- Ticket: https://torchbox.monday.com/boards/1192293412/views/4019870/pulses/1193036756?term=image
- I have not followed a design for this

Overview of changes
- Removes the slightly crude overrides of theme colours that were previously done for all proposition pages, and instead adds some extra theme variables to handle the fact that the dark theme has links that sit both on the dark header area, and on the white body of the page.
- Removes the option to set a proposition page with the 'dark transparent' theme, as this has never worked.

The main aim of this ticket was to ensure that the 'dark' theme used the light blue colours for accents - see this preview of an unpublished service page on the live site to see how this previously looked for service pages: https://torchbox.com/admin/pages/1196/edit/preview/?mode=

In testing though I found a number of issues with different components in different themes which I have addressed as part of this ticket.

To test:
The pattern library page (http://localhost:8000/pattern-library/render-pattern/patterns/pages/proposition/proposition.html) uses the light theme, and does not show all the elements, so it is best to create a new wagtail proposition page (or edit an existing one), and test the following items:
- Links in the intro field (in the hero) for all themes (these have changed a bit in order to create a hover effect)
- In page nav links in the hero and sticky header in all themes
- Links in intro fields for 'our services', 'our clients' etc
- Key points headings in all themes
- CTA without an embed in all themes, with a span tag to change the colour
- Check the hover state on the CTA button doesn't add a pale blue bottom border
- CTA *with* an embed in all themes, with a span tag to change the colour
- Check the hover state on the CTA with embed button doesn't add a pale blue bottom border

### CODE REVIEW

**Comment by helenb** [comment]
@engineervix would you mind reviewing my model change here? We should maybe create a data migration in case any proposition pages have the 'dark transparent' theme set, but I think it is unlikely any would as it doesn't work. If you think it needs a migration, it would be great if you could give me some advice on how to go about it.

**Comment by engineervix** [comment]
> @engineervix would you mind reviewing my model change here? We should maybe create a data migration in case any proposition pages have the 'dark transparent' theme set, but I think it is unlikely any would as it doesn't work. If you think it needs a migration, it would be great if you could give me some advice on how to go about it.

Hi @helenb I've had a look at the model change and I think a data migration will be necessary, since existing Pages with the "dark--transparent" theme will not be automatically updated, and this may result in `ValueError`s.

Here's how I'd go about it:

1. Create a new data migration before removing the option.

  ```shell
  python manage.py makemigrations propositions --name data_migration_proposition_theme --empty 
  ```

  ```python
  from django.db import migrations
  
  
  def update_dark_transparent_theme(apps, schema_editor):
      Page = apps.get_model('propositions', 'PropositionPage')
      Page.objects.filter(theme='dark--transparent').update(theme='light')
  
  
  class Migration(migrations.Migration):
  
      dependencies = [
          ('propositions', '0009_auto_20230503_1527'),
      ]
  
      operations = [
          migrations.RunPython(update_dark_transparent_theme),
      ]
  ```

2. Run the migration to update the existing data.

  ```shell
  python manage.py migrate propositions
  ```

3. Remove the option from the model.
4. Generate a new migration after removing the option

**Comment by helenb** [comment]
> > @engineervix would you mind reviewing my model change here? We should maybe create a data migration in case any proposition pages have the 'dark transparent' theme set, but I think it is unlikely any would as it doesn't work. If you think it needs a migration, it would be great if you could give me some advice on how to go about it.
> 
> > Note
> > Just realized that this seems to have already been done in #477

Ah I didn't see this comment in time and I've now pushed it to this update. I think we could actually close 477 now and just have the change here.

**Comment by helenb** [comment]
@engineervix if #477 was already on staging will this not work when deployed as it assumes the dark theme is in place when running the data migration?

File: tbx/static_src/sass/components/_key-points.scss [hunk 1]
```diff
@@ -28,6 +28,11 @@
                 border-bottom: 0;
             }
         }
+
+        &::before {
+            //overde underline colour in mixin
```
Comment (albinazs) [inline tbx/static_src/sass/components/_key-points.scss:33]: override?

**Review by albinazs — APPROVED** [review]
@helenb, looks good to me!

File: tbx/static_src/sass/components/_key-points.scss (hunk 1 above)
Comment (helenb) [reply tbx/static_src/sass/components/_key-points.scss:33]: oops, will amend!

**Pull request merged**